runtime/bindings/python/io.cc

```diff
@@ -178,8 +216,98 @@ void SetupIoBindings(py::module_ &m) {
                                         writable, unused_len);
           },
           py::arg("host_buffer"), py::arg("readable") = true,
-          py::arg("writable") = false);
+          py::arg("writable") = false)
+      .def_prop_ro(
+          "is_host_allocation",
+          [](FileHandle &self) {
+            auto primitive = iree_io_file_handle_primitive(self.raw_ptr());
+            return primitive.type == IREE_IO_FILE_HANDLE_TYPE_HOST_ALLOCATION;
+          })
+      .def_prop_ro(
+          "host_allocation",
+          [](py::handle self) {
+            return py::steal<py::object>(PyMemoryView_FromObject(self.ptr()));
+          })
+      .def("__repr__", [](py::handle self_object) {
+        if (py::cast<py::bool_>(self_object.attr("is_host_allocation"))) {
```

Comment:
I'm seeing build errors here on Windows with Python 3.11:

```
[build] D:\dev\projects\iree\runtime\bindings\python\io.cc(232): error C2039: 'bool_': is not a member of 'nanobind'
[build] D:\dev\projects\iree\.venv\Lib\site-packages\nanobind\include\nanobind/stl/vector.h(15): note: see declaration of 'nanobind'
[build] D:\dev\projects\iree\runtime\bindings\python\io.cc(232): error C2065: 'bool_': undeclared identifier
```

CI seems able to build though...

Reply:
Ah. I need to download a new version from the requirements, that's not in a submodule

Reply:
Sent https://github.com/openxla/iree/pull/16976 to help with that